.github/workflows/binary-build.yml

```diff
@@ -0,0 +1,162 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+name: Build binary and container for single arch
+
+permissions:
+  contents: read
+
+on:
+  workflow_call:
+    inputs:
+      publishType:
+        required: true
+        type: string
+      arch:
+        required: true
+        type: string
+
+env:
+  EXPECTED_GO_VERSION: "1.24.1"
```

Comment:
where does the go version come from? Is it pre-installed?

Reply:
This matches the version in the `go.mod` file. This helps prevent the Go version from being updated accidentally, since you have to update it here as well.